rotkehlchen/api/v1/schemas.py

```diff
@@ -3127,3 +3127,79 @@ class ClearAvatarsCacheSchema(Schema):
         fields.String(required=True, validate=lambda x: x.endswith('.eth')),
         load_default=None,
     )
+
+
+class EthStakingHistoryStats(AsyncQueryArgumentSchema, OnlyCacheQuerySchema):
+    """Schema for quering history events"""
+    from_timestamp = TimestampField(load_default=Timestamp(0))
+    to_timestamp = TimestampField(load_default=ts_now)
+    location_labels = DelimitedOrNormalList(fields.String(), load_default=None)
+    validator_indices = DelimitedOrNormalList(fields.Integer(), load_default=None)
+
+    def __init__(self, db: 'DBHandler') -> None:
+        super().__init__()
+        self.db = db
+
+    @post_load
+    def make_history_event_filter(
+            self,
+            data: dict[str, Any],
+            **_kwargs: Any,
+    ) -> dict[str, Any]:
+        base_arguments = {
+            'from_ts': data['from_timestamp'],
+            'to_ts': data['to_timestamp'],
+        }
+        validator_indices = data['validator_indices']
+
+        # for mev rewards if we filter by address is possible to miss the block production
+        # since the locaiton label is the builder address and not the user address. To avoid this
+        # we use the location label to get the validator index and then use it to add extra
+        # validator indices to not miss any event.
+        if (location_labels := data['location_labels']) is not None:
+            query, bindings = EthStakingEventFilterQuery.make(
+                **base_arguments,
+                location_labels=location_labels,
+                event_types=[HistoryEventType.STAKING],
+                event_subtypes=[HistoryEventSubType.MEV_REWARD],
+                entry_types=[HistoryBaseEntryType.ETH_BLOCK_EVENT],
+            ).prepare(with_pagination=False)
+            base_query = (
+                'SELECT validator_index FROM history_events JOIN '
+                'eth_staking_events_info ON eth_staking_events_info.identifier='
+                'history_events.identifier '
+            )
+            with self.db.conn.read_ctx() as cursor:
+                extra_indices = [entry[0] for entry in cursor.execute(base_query + query, bindings)]  # noqa: E501
+            if validator_indices is not None and len(extra_indices) != 0:
+                validator_indices.extend(extra_indices)
+
+        if None in (location_labels, validator_indices):
+            common_arguments = base_arguments | {
+                'location_labels': location_labels,
+                'validator_indices': validator_indices,
+            }
+        else:
+            common_arguments = base_arguments | {
```

Comment:
As said in the comment in `filtering.py`, I think it will be much cleaner to handle this condition directly in the filtering class. So always pass `location_labels` and `validator_indices`, and if they are both not None, make the nested `or` filter.

Reply:
We discussed about this in the discord thread https://discord.com/channels/657906918408585217/1102958606061686954/1102959668776677376 And I prefer to not move the logic to the filter itself. The reason is that you would still need an argument to tell if this two fields should be queried with an `AND` or an `OR`, that adds a db query inside the filter and this is bad imo, and we are abstracting in a way that maybe is not needed anywhere else. If we endup needing it we can then move to the filter